test/data/javascript.js

```diff
@@ -104,6 +107,218 @@ exports.test_data = {
 
         ],
     }, {
+        name: "operator_position option - ensure no neswlines if preserve_newlines is false",
+        matrix: [
+            {
+                options: [
+                    { name: "operator_position", value: "'before_newline'" },
+                    { name: "preserve_newlines", value: "false" }
+                ]
+            }, {
+                options: [
+                    { name: "operator_position", value: "'after_newline'" },
+                    { name: "preserve_newlines", value: "false" }
+                ]
+            }, {
+                options: [
+                    { name: "operator_position", value: "'preserve_newline'" },
+                    { name: "preserve_newlines", value: "false" }
+                ]
+            }
+        ],
+        tests: [
+            {
+                unchanged: inputlib.operator_position.sanity
+            }, {
+                input: inputlib.operator_position.comprehensive,
+                output: inputlib.operator_position.sanity,
+            }
+        ]
+    }, {
+        name: "reset preserve_newlines and operator_position",
+        options: [
+            { name: "preserve_newlines", value: "true" },
+            { name: "operator_position", value: "'before_newline'" }
+        ]
+    }, {
+        name: "operator_position option - set to 'before_newline' (default value)",
+        tests: [
+            {
+                comment: 'comprehensive, various newlines',
+                input: inputlib.operator_position.comprehensive,
+                output: [
+                    'var res = a + b - c /',
+                    '    d * e %',
+                    '    f;',
```

Comment:
> Then people will expect the following:

That is exactly what they should expect for `after_newline`.  If they want to preserve the existing newline positioning, they'll use `preserve_newline`.  

I think you're right, it will involve either waiting until the next token to see if it wants a newline and so on.  Messy, but so it goes.   Or you could peek ahead, since all the tokens are already available.